src/imagecapturecore.cs

```diff
@@ -0,0 +1,901 @@
+		NSString CameraDeviceCanAcceptPtpCommands { get; }
+	}
+
+	[Static]
+	interface ICScannerStatus {
+		[Field ("ICScannerStatusWarmingUp")]
+		NSString WarmingUp { get; }
+
+		[Field ("ICScannerStatusWarmUpDone")]
+		NSString WarmUpDone { get; }
+
+		[Field ("ICScannerStatusRequestsOverviewScan")]
+		NSString RequestsOverviewScan { get; }
+	}
+
+	[Static]
+	interface ICDeviceLocationDescriptions {
+		[Field ("ICDeviceLocationDescriptionUSB")]
+		NSString Usb { get; }
+
+		[Field ("ICDeviceLocationDescriptionFireWire")]
+		NSString FireWire { get; }
+
+		[Field ("ICDeviceLocationDescriptionBluetooth")]
+		NSString Bluetooth { get; }
+
+		[Field ("ICDeviceLocationDescriptionMassStorage")]
+		NSString MassStorage { get; }
+	}
+
+	[Static]
+	interface ICCameraDownloadOptionKeys {
+		[Field ("ICDownloadsDirectoryURL")]
+		NSString DownloadsDirectoryUrl { get; }
+
+		[Field ("ICSaveAsFilename")]
+		NSString SaveAsFilename { get; }
+
+		[Field ("ICSavedFilename")]
+		NSString SavedFilename { get; }
+
+		[Field ("ICSavedAncillaryFiles")]
+		NSString SavedAncillaryFiles { get; }
+
+		[Field ("ICOverwrite")]
+		NSString Overwrite { get; }
+
+		[Field ("ICDeleteAfterSuccessfulDownload")]
+		NSString DeleteAfterSuccessfulDownload { get; }
+
+		[Field ("ICDownloadSidecarFiles")]
+		NSString DownloadSidecarFiles { get; }
+	}
+
+	[Static]
+	interface ICStatusNotificationKeys { // Keys into DidReceiveStatusInformation dictionary 
+		[Field ("ICStatusNotificationKey")]
+		NSString NotificationKey { get; }
+
+		[Field ("ICStatusCodeKey")]
+		NSString CodeKey { get; }
+
+		[Field ("ICLocalizedStatusNotificationKey")]
+		NSString LocalizedNotificationKey { get; }
+	}
+
+	[Static]
+	interface ICButtonType {
+		[Field ("ICButtonTypeScan")]
+		NSString Scan { get; }
+
+		[Field ("ICButtonTypeMail")]
+		NSString Mail { get; }
+
+		[Field ("ICButtonTypeCopy")]
+		NSString Copy { get; }
+
+		[Field ("ICButtonTypeWeb")]
+		NSString Web { get; }
+
+		[Field ("ICButtonTypePrint")]
+		NSString Print { get; }
+
+		[Field ("ICButtonTypeTransfer")]
+		NSString Transfer { get; }
+	}
+
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof(NSObject))]
+	interface ICDeviceDelegate {
+
+		[Abstract]
+		[Export ("didRemoveDevice:")]
+		void DidRemoveDevice (ICDevice device);
+	
+		[Export ("device:didOpenSessionWithError:")]
+		void DidOpenSession (ICDevice device, [NullAllowed] NSError error);
+	
+		[Export ("deviceDidBecomeReady:")]
+		void DidBecomeReady (ICDevice device);
+	
+		[Export ("device:didCloseSessionWithError:")]
+		void DidCloseSession (ICDevice device, [NullAllowed] NSError error);
+	
+		[Export ("deviceDidChangeName:")]
+		void DidChangeName (ICDevice device);
+	
+		[Deprecated (PlatformName.MacOSX, 10, 13)]
+		[Export ("deviceDidChangeSharingState:")]
+		void DidChangeSharingState (ICDevice device);
+	
+		[Export ("device:didReceiveStatusInformation:")]
+		void DidReceiveStatusInformation (ICDevice device, NSDictionary<NSString, NSObject> status);
```

Comment:
We could make the dictionary to me a strong one. If we do not want to do it this time, it will be better to use a NSDictionary to later have the chance to use the strong dict.

Reply:
Unfortunately we can't do strong dictionaries for members inside protocols